server/services/offendersService.js

```diff
@@ -144,19 +135,31 @@ module.exports = function createOffendersService(nomisClientBuilder, allocationC
       }
 
       const combined = [...uncategorisedResult, ...dbManualInProgress]
-      const [sentenceMap, offenceMap, pomMap, securityReferredOffenders] = await Promise.all([
-        getSentenceMap(combined, nomisClient),
-        getOffenceMap(combined, nomisClient),
+
+      const bookingIds = combined
+        .filter(o => !o.dbRecord || !o.dbRecord.catType || o.dbRecord.catType === CatType.INITIAL.name)
+        .map(o => o.bookingId)
+
+      const [prisoners, pomMap, securityReferredOffenders] = await Promise.all([
+        prisonerSearchClient.getPrisonersByBookingIds(bookingIds),
         getPomMap(combined, allocationClient),
         formService.getSecurityReferrals(agencyId, transactionalDbClient),
       ])
 
+      const prisonerMap = new Map(prisoners.map(prisoner => [prisoner.bookingId, prisoner]))
+
+      const sentenceMap = new Map(
+        prisoners
+          .filter(s => s.sentenceStartDate) // the endpoint returns records for offenders without sentences
+          .map(s => [s.bookingId, { sentenceDate: s.sentenceStartDate }])
+      )
+
```

Comment:
getPrisonersByBookingIds() gets the data for both maps. In fact 2 separate maps are not really needed either